event-session-signup/Code.gs

```diff
@@ -0,0 +1,176 @@
+/**
+ * A special function that inserts a custom menu when the spreadsheet opens.
+ */
+function onOpen() {
+  SpreadsheetApp.getUi().createMenu('Conference')
+    .addItem('Set up conference', 'setUpConference_')
+    .addToUi();
+}
+
+/**
+ * A set-up function that uses the conference data in the spreadsheet to create
+ * Google Calendar events, a Google Form, and a trigger that allows the script
+ * to react to form responses.
+ */
+function setUpConference_() {
+  if (ScriptProperties.getProperty('calId')) {
+    Browser.msgBox('Your conference is already set up. Look in Google Drive!');
+  }
+  var ss = SpreadsheetApp.getActive();
+  var sheet = ss.getSheetByName('Conference Setup');
+  var range = sheet.getDataRange();
+  var values = range.getValues();
+  setUpCalendar_(values, range);
+  setUpForm_(ss, values);
+  ScriptApp.newTrigger('onFormSubmit').forSpreadsheet(ss).onFormSubmit()
+      .create();
+  ss.removeMenu('Conference');
```

Comment:
Won't this just get re-added next time sheet is opened? If so, might be better to just leave it here, otherwise might be confusing to somebody trying to understand this -- weird if they remove the menu, but then it shows up again...

Reply:
+1 
it was a difficult flow to understand, especially when wanting to re-do the whole workflow in the same spreadsheet